src/main/java/com/adaptavist/tm4j/jenkins/extensions/Instance.java

```diff
@@ -0,0 +1,23 @@
+package com.adaptavist.tm4j.jenkins.extensions;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+import java.io.File;
+
+public interface Instance {
```

Comment:
Don't you think it should be called `JiraInstance`?

Reply:
Yes, I'd make more sense to be called `JiraInstance` and have `JiraServerInstance` and `JiraCloudInstance`, but because Jenkins saves the configuration in an XML file looking like 

`<com.adaptavist.tm4j.jenkins.extensions.configuration.Tm4jGlobalConfiguration plugin="tm4j-automation@1.0.0">
  <jiraInstances>
    <com.adaptavist.tm4j.jenkins.extensions.JiraInstance>
      <serverAddress>http://localhost:2990/jira</serverAddress>
      <username>admin</username>
      <password>admin</password>
    </com.adaptavist.tm4j.jenkins.extensions.JiraInstance>
  </jiraInstances>
</com.adaptavist.tm4j.jenkins.extensions.configuration.Tm4jGlobalConfiguration>`

That will then get deserialised in the specific beans. Renaming the `JiraInstance`, which was used as the bean for server, into `JiraServerInstance` will invalidate all previous configurations.

Currently, the XML looks like
`<com.adaptavist.tm4j.jenkins.extensions.configuration.Tm4jGlobalConfiguration plugin="tm4j-automation@1.1.0">
  <jiraInstances>
    <com.adaptavist.tm4j.jenkins.extensions.JiraCloudInstance>
      <jwt>token</jwt>
      <name>URL</name>
    </com.adaptavist.tm4j.jenkins.extensions.JiraCloudInstance>
    <com.adaptavist.tm4j.jenkins.extensions.JiraInstance>
      <serverAddress>URL</serverAddress>
      <username>admin</username>
      <password>admin</password>
    </com.adaptavist.tm4j.jenkins.extensions.JiraInstance>
  </jiraInstances>
</com.adaptavist.tm4j.jenkins.extensions.configuration.Tm4jGlobalConfiguration>`

Reply:
Very well observed!